### ISSUE TITLE
AutoCombi

### ISSUE THREAD

**Issue by Paul2510** [Mogli12/AutoCombine#18]
Hi! Hoffe das dieses super Tool auch in Ls 17 kommt! Deine Sachen sind wirklich super! Danke für deine tolle Arbeit!

**Comment by Mogli12** [comment]
Hi,

In LS15 gab es die Spezifikationen aiTractor und aiCombine. Beide wurden durch aiVehicle ersetzt. Diese beinhaltet nun nicht mehr die eigentlichen Fahrmanöver. Diese sind in verschiedene AIDriveStrategy... Klassen ausgelagert und gerade diese fehlen in der [ScriptDocu](https://gdn.giants-software.com/documentation_scripting.php).

=> Es wird wohl weder AutoTractor noch AutoCombine in FS17 geben. 

Ich werde mir selber den Kursgenerator in Courseplay genauer anschauen. Dort kann man sich ja schon in LS15 einen Kurs für ein Feld generieren lassen.

Viele Grüße

**Comment by Mogli12** [comment]
Hi,

In LS15, there were the specifications aiTractor and aiCombine. Both were replaced by aiVehicle. This does not include the actual driving maneuvers. These are implemented in various AIDriveStrategy ... classes and these are missing in the [script docu](https://gdn.giants-software.com/documentation_scripting.php).

=> There will probably be neither AutoTractor nor AutoCombine in FS17.

I will take a closer look at the course generator in Courseplay. There you can already have a course for a field generated in LS15.

Many greetings

**Comment by Fogest** [comment]
Hey @Mogli12 if you're going to use a system like Courseplay I don't see why people wouldn't just use Courseplay then?

**Comment by Mogli12** [comment]
Hi, my proposal is to use Courseplay in FS17. AutoCombine will not be converted to FS17.

**Comment by Fogest** [comment]
@Mogli12 Sounds good, I am part of the Courseplay group and in the code we have some integration from  AutoCombine in it to support it so for FS17 that support can be removed if AutoCombine won't be updated.

**Comment by genen** [comment]
Having AutoCombine follow the field border of an unusual field was much easier with multiple combines than trying to program Courseplay with 2 or more combines to do one field following its contours - if not impossible.

Sorry to see the product retired without a replacement option

**Comment by Fogest** [comment]
@genen Without a replacement option? Courseplay is the replacement option. Maybe not the most ideal one in your opinion right now, however it is still being worked on and improved, and especially with AutoCombine being removed there may be more of a gap to fill. 

Additionally you can define custom fields in Courseplay so irregular custom fields should not be an issue. There are some precision issues with sometimes small bits of crops being missed however it is constantly being worked on and should hopefully be eventually addressed :).

Hopefully the gap you feel is missing can eventually be filled! Feel free to make suggestions toward the courseplay team on their Github page.

**Comment by genen** [comment]
I used courseplay for 15 and plan to in 17. However for running multiple combines on a field, auto combine had fewer issues than courseplay with stalled equipment. The other issue was on headlands courseplay left areas untouched in planting and harvesting. Auto combine sensed the fruit layer and adjusted in real time to cut following the edge of it. It also did tight turns better than built in AI with either omega turns or 3 point turns. Courseplay does much more, but colisions and crop sensing were better with auto combine.  

The other problem with courseplay is that it probably won't be out for months. I'm on the courseplay github looking at progress and it's a complex set of code with much to update and test. 

I'm a moder on Unreal engine, I get the it takes a long time to release a bug free mod, so with both simpler mods (GPS and followme) now out, it helps with 17s absolutely crappy built in AI which decides to change direction mid field. Auto combine would add a much needed alternative to the broken AI and the delay of courseplay release. 

If I had the free time I would happily branch off this and develop a 17 version. I've tried a couple simple aiVehicle tests and am figuring it out, but don't have much time.

**Comment by Fogest** [comment]
May I make a suggestion that you contribute to Courseplay then? AutoCombine would require a lot more work to get it working in FS17 and would end up using similar mechanisms to Courseplay. I am apart of the team for CP and a lot of Courseplay is working with FS17 already. There are still some kinks that need to e worked out but I can't see it taking months for it to be released.

**Comment by genen** [comment]
I could...  time is the limiting factor for me right now in general... but I'd be happy to help in whatever way I can. Point me where to get started with the team there.

**Comment by Fogest** [comment]
Helping with the update to FS17 is not something you'll be able to do much for, however once that is released you should be able to see the code and help make a pull request if you want to try and improve the combine. I'd suggest before doing that to create an issue on the github first. One of the developers from the team posted an update on a github issue with the progress so you can see that it is nearing being ready.

**Comment by SirAlbertL** [comment]
One thing I have been able to do, is record my OWN field course...{driving it out complacently}. Save it. and when I go and send the AI to do that same job, it dose it perfectly.

Now I have noticed that the Features in FS17 can read each field, even partially worked fields and they show up clearly on the "hud/pda" map. 

The system can detect how long it has been since you plowed... in cultivation register growth stages, fruit types and even three stages of fertilization.

Surely if the engine is able to detect, read all of this. Certainly there must be a way to read the these in the tractor or combine and apply it to it in real-time to maintaining position, and fallow the working line.


But I am a farmer.... and can only do very basic changes/or even understand lua scripts on a very basic level.
